core/storage/question/gae_models.py

```diff
@@ -450,21 +483,35 @@ def get_question_skill_links_equidistributed_by_skill(
         question_skill_link_models = []
         existing_question_ids = []
 
+        def get_offset(query):
+            """Helper function to get the offset."""
+            question_count = query.count()
+            if question_count > 2 * question_count_per_skill:
+                return utils.get_random_int(
+                    question_count - (question_count_per_skill * 2))
+            return 0
+
         for skill_id in skill_ids:
             query = cls.query(cls.skill_id == skill_id)
+
             # We fetch more questions here in order to try and ensure that the
             # eventual number of returned questions is sufficient to meet the
             # number requested, even after deduplication.
             new_question_skill_link_models = query.fetch(
-                question_count_per_skill * 2)
+                limit=question_count_per_skill * 2,
+                offset=get_offset(query))
 
             # Deduplicate if the same question is linked to multiple skills.
             for model in new_question_skill_link_models:
                 if model.question_id in existing_question_ids:
                     new_question_skill_link_models.remove(model)
+            if len(new_question_skill_link_models) > question_count_per_skill:
+                new_question_skill_link_models = utils.get_random_sample(
```

Comment:
Do not redefine variables. Choose a different name instead.

Reply:
Done.